src/Module.php

```diff
@@ -104,10 +108,17 @@ public function add_zone_backend(string $domain, string $ip): bool
 			 */
 			try
 			{
+				$kind = 'native'; // Enables backend replication via MySQL Replication or MariaDB Galera replication
+
+				if (defined('AUTH_PDNS_TYPE') && in_array(AUTH_PDNS_TYPE, static::$permitted_types))
+				{
+					$kind = AUTH_PDNS_TYPE;
+				}
+
 				$nsNames = $this->get_hosting_nameservers($domain);
 				$api = $this->makeApi($domain);
 				$api->do('POST', 'servers/localhost/zones', [
-					'kind'        => 'native', // Enables backend replication via MySQL Replication or MariaDB Galera replication
+					'kind'        => $kind,
```

Comment:
For clarity add a class constant,

// @var string default PowerDNS server type
protected const PDNS_SERVER_TYPE = 'native';

then it could be rewritten as

`'kind' => $kind ?? PDNS_SERVER_TYPE`

The reader is a little more informed when looking over the code how PowerDNS behaves in a bare environment rather than jumping down to the bottom.